pom.xml

```diff
@@ -1,80 +1,219 @@
+                        <annotationProcessorPaths>
+                            <path>
+                                <groupId>org.projectlombok</groupId>
+                                <artifactId>lombok</artifactId>
+                                <version>${lombok.version}</version>
+                            </path>
+                            <annotationProcessorPath> <!-- necessary to generate META-INF/spring-configuration-metadata.json -->
+                                <groupId>org.springframework.boot</groupId>
+                                <artifactId>spring-boot-configuration-processor</artifactId>
+                                <version>2.4.4</version>
+                            </annotationProcessorPath>
+                        </annotationProcessorPaths>
+                    </configuration>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+        <resources>
+            <resource>
+                <directory>src/main/resources</directory>
+                <filtering>true</filtering>
+            </resource>
+        </resources>
+        <plugins>
+            <plugin>
+                <groupId>org.springframework.boot</groupId>
+                <artifactId>spring-boot-maven-plugin</artifactId>
+            </plugin>
+            <plugin>
+                <groupId>org.projectlombok</groupId>
+                <artifactId>lombok-maven-plugin</artifactId>
+                <version>${lombok.version}.0</version>
+                <executions>
+                    <execution>
+                        <phase>generate-sources</phase>
+                        <goals>
+                            <goal>delombok</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+
+    <profiles>
+        <profile>
+            <id>default</id>
+            <activation>
+                <activeByDefault>true</activeByDefault>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-compiler-plugin</artifactId>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+        <profile>
+            <id>java16</id>
+            <activation>
+                <jdk>16</jdk>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-compiler-plugin</artifactId>
+                        <configuration>
+                            <release>${jdk.release}</release>
+                            <!-- Below is needed to make lombok work with JDK16.  See https://github.com/rzwitserloot/lombok/issues/2681 -->

```

Comment:
The referenced issue is now closed, I think this configuration is not needed for the latest version of lombok, 1.18.20

Reply:
I can simplify this now that JDK17 is out.  Looks like I was using preview language features in JDK16 when I did this.  I'll play around with it.

Reply:
Thank you so much!

Reply:
OK, I did that.  This works for me with JDK17.  I don't have JDK16 anymore.  It'd need more work to make this work with multiple JDK versions.  I don't have 9, but I'm guessing it won't build in 9 now without some `pom.xml` changes.

Reply:
I tested with 11 and 17.

Reply:
Awesome, just let me give it a final check before merging. Java 11 and 17 are ok, I'll create another branch with the older code. Thanks!